### ISSUE TITLE
java.util.ConcurrentModificationException

### ISSUE THREAD

**Issue by konikvranik** [revapi/revapi#250]
Revapi fails with `java.util.ConcurrentModificationException`

```
Caused by: java.util.ConcurrentModificationException
	at java.util.ArrayList$Itr.checkForComodification(ArrayList.java:911)
	at java.util.ArrayList$Itr.next(ArrayList.java:861)
	at org.revapi.Revapi.transform(Revapi.java:598)
	at org.revapi.Revapi.lambda$analyzeWith$6(Revapi.java:350)
	at java.util.ArrayList.forEach(ArrayList.java:1259)
	at org.revapi.Revapi.analyzeWith(Revapi.java:349)
	at org.revapi.Revapi.analyze(Revapi.java:212)
	at org.revapi.Revapi$analyze$0.call(Unknown Source)
```

**Comment by konikvranik** [comment]
This happens randomly.
I call directly the revapi API from our gradle plugin.
We have parallelism enabled for gradle.
My dependencies for the gradle plugin:

- 'org.revapi:revapi-basic-features:0.12.3',
-			'org.revapi:revapi-java:0.25.0',
-			'org.revapi:revapi-reporter-text:0.14.3',
-			'org.revapi:revapi-reporter-json:0.4.3',

**Comment by metlos** [comment]
Btw. I've noticed you mentioned in some other issue that you use your own custom gradle plugin. I already put in the fix in the `main` branch of Revapi (b3fc4b0f112bf293e342214665e54e518e9288a5) so if you build the `main` of Revapi (`mvn install -Pfast`) and use those snapshot versions in your gradle plugin, you should no longer see those CMEs.

**Comment by metlos** [comment]
Fixed in `revapi:0.14.4`.

**Comment by konikvranik** [comment]
I didn't see the issue for a long time, so it looks to be fixed. Thank you!
